pkg/issuer/acme/setup.go

```diff
@@ -68,51 +69,121 @@ func (a *Acme) Setup(ctx context.Context) (issuer.SetupResponse, error) {
 		ns = a.IssuerOptions.ClusterResourceNamespace
 	}
 
-	// attempt to obtain the existing private key, if it does not exist then
-	// we generate one
+	// attempt to obtain the existing private key from the apiserver.
+	// if it does not exist then we generate one
+	// if it contains invalid data, warn the user and return without error.
+	// if any other error occurs, return it and retry.
 	pk, err := a.helper.ReadPrivateKey(a.issuer.GetSpec().ACME.PrivateKey, ns)
-	if err != nil {
-		if !k8sErrors.IsNotFound(err) && !errors.IsInvalidData(err) {
-			s := messageAccountVerificationFailed + err.Error()
-			a.issuer.UpdateStatusCondition(v1alpha1.IssuerConditionReady, v1alpha1.ConditionFalse, errorAccountVerificationFailed, s)
-			return issuer.SetupResponse{Requeue: true}, fmt.Errorf(s)
-		}
-
+	switch {
+	case apierrors.IsNotFound(err):
 		glog.Infof("%s: generating acme account private key %q", a.issuer.GetObjectMeta().Name, a.issuer.GetSpec().ACME.PrivateKey.Name)
 		pk, err = a.createAccountPrivateKey(a.issuer.GetSpec().ACME.PrivateKey, ns)
 		if err != nil {
 			s := messageAccountRegistrationFailed + err.Error()
 			a.issuer.UpdateStatusCondition(v1alpha1.IssuerConditionReady, v1alpha1.ConditionFalse, errorAccountRegistrationFailed, s)
 			return issuer.SetupResponse{Requeue: true}, fmt.Errorf(s)
 		}
-
+		// We clear the ACME account URI as we have generated a new private key
 		a.issuer.GetStatus().ACMEStatus().URI = ""
+
+	case errors.IsInvalidData(err):
+		a.issuer.UpdateStatusCondition(v1alpha1.IssuerConditionReady, v1alpha1.ConditionFalse, errorAccountVerificationFailed, fmt.Sprintf("Account private key is invalid: %v", err))
```

Comment:
We no longer generate a new private key if the existing private key is 'invalid' (instead, we warn the user so they can take manual action)